tests/unit/s2n_async_pkey_test.c

```diff
@@ -300,7 +426,66 @@ int main(int argc, char **argv)
 
     EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
 
+    struct s2n_hash_state digest = { 0 };
+    EXPECT_SUCCESS(s2n_hash_new(&digest));
+    EXPECT_SUCCESS(s2n_hash_init(&digest, S2N_HASH_SHA1));
+    EXPECT_SUCCESS(s2n_hash_update(&digest, test_digest_data, test_digest_size));
+
+    /* Test: signature offload. */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+        conn->config->async_pkey_cb = async_pkey_signature_callback;
+
+        EXPECT_FALSE(s2n_result_is_ok(s2n_async_pkey_sign(conn, S2N_SIGNATURE_ECDSA, &digest, s2n_async_sign_complete)));
+        EXPECT_TRUE(s2n_errno == S2N_ERR_ASYNC_BLOCKED);
```

Comment:
Here and below, if the "async" call is actually synchronous, why does this call set `S2N_ERR_ASYNC_BLOCKED`?

Reply:
In this case the async statemachine is set to blocked, so the user can poll the handshake. The handshake is no longer in the `S2N_ERR_ASYNC_BLOCKED` state when `s2n_async_pkey_op_apply` is called. 

The `s2n_async_pkey_op_*` does not allow for the operation to be completed before the callback returns, so the work just needs to be done outside of the callback.